docusaurus-plugin-moonwave/src/components/LuaType.js

```diff
@@ -181,18 +193,46 @@ function Token({ token, depth }) {
     case "union":
       return <Op>&nbsp;|&nbsp;</Op>
     case "luaType":
+      const sanitizedToken = token.luaType.replace(/\W/g, "")
+      // Checks if the type is in the list of typeLinks for the package
+      if (typeLinks.has(sanitizedToken)) {
+        // If the type is to an external Roblox default, link to the external Roblox Devhub
+        if (isValidUrl(typeLinks.get(sanitizedToken))) {
+          return (
+            <code className={styles.blue}>
+              <a
+                style={{ textDecoration: "underline", color: "inherit" }}
+                href={typeLinks.get(sanitizedToken)}
+              >
+                {token.luaType}
+              </a>
+            </code>
+          )
+        }
+        // If the type is from another page in the Moonwave package
+        else {
+          return (
+            <code className={styles.blue}>
+              <Link
+                style={{ textDecoration: "underline", color: "inherit" }}
+                to={typeLinks.get(sanitizedToken)}
+              >
+                {token.luaType}
+              </Link>
+            </code>
+          )
+        }
+      }
+
```

Comment:
Instead of branching and writing all this code twice, we should extract the `Link` and `a` elements into a new component, maybe called `GenericLink`, which simply checks if the URL starts with `https?://` and then returns either an `a` element or a `Link` element. Then, you should just write this code once and use the `GenericLink` component.